src/repl-frontend/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -78,10 +78,51 @@ class MainGenericRunner {
           None
         case _  =>
           // We start the repl when no arguments are given.
-          // If user is agnostic about both -feature and -deprecation, turn them on.
+          // If user is agnostic about both -feature and -deprecation, turn them on and add batteries.
           if (settings.deprecation.isDefault && settings.feature.isDefault && settings.lint.isDefault) {
             settings.deprecation.value = true
             settings.feature.value = true
+            locally {
+              import settings.LintWarnings._
+              for (linted <- allLintWarnings if linted != Unused)
+                settings.lint.enable(linted)
+            }
```

Comment:
@lrytz sometimes I think the multichoice setting is clunky, but look ma, no strings!